In [19]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score, classification_report

# Load the dataset
df = pd.read_csv(r'C:\Users\Aksha\Downloads\Project Heart\heart-resting 0 eliminated.csv')

# Preprocess the data
X = df.drop('HeartDisease', axis=1)
y = df['HeartDisease']

# Convert categorical variables to dummy variables
X = pd.get_dummies(X, drop_first=True)

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Standardize the features
scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

# Train the KNN classifier
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(X_train, y_train)

# Make predictions
y_pred = knn.predict(X_test)

# Evaluate the model
accuracy = accuracy_score(y_test, y_pred)
print(f'Accuracy: {accuracy}')
print('Classification Report:')
print(classification_report(y_test, y_pred))

Accuracy: 0.8641304347826086
Classification Report:
              precision    recall  f1-score   support

           0       0.81      0.86      0.83        72
           1       0.91      0.87      0.89       112

    accuracy                           0.86       184
   macro avg       0.86      0.86      0.86       184
weighted avg       0.87      0.86      0.86       184




- **Accuracy: 0.8641304347826086**
  - This means that the model correctly classified approximately 86.41% of the instances in the test set.

The KNN model performs well with an overall accuracy of 86.41%. The precision, recall, and F1-scores are also high, indicating that the model is effective at distinguishing between the two classes (heart disease present or not). The metrics suggest that the model is slightly better at predicting class 1 (heart disease present) than class 0.